packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol

```diff
@@ -133,6 +220,24 @@ contract BullStrategyTestFork is Test {
         }     
     }
 
+    function _calcWPowerPerpAndCrabNeededForWithdraw(uint256 _bullAmount) internal view returns (uint256, uint256) {
+        uint256 share = _bullAmount.wdiv(bullStrategy.totalSupply());
+        uint256 crabToRedeem = share.wmul(IERC20(crabV2).balanceOf(address(bullStrategy)));
+        uint256 crabTotalSupply = IERC20(crabV2).totalSupply();
+        (, uint256 squeethInCrab) = _getCrabVaultDetails();
+        return (crabToRedeem.wmul(squeethInCrab).wdiv(crabTotalSupply), crabToRedeem);
+    }
+
+    function _calcUsdcNeededForWithdraw(uint256 _bullAmount) internal view returns (uint256) {
+        uint256 share = _bullAmount.wdiv(bullStrategy.totalSupply());
+        return share.wmul(IEulerDToken(dToken).balanceOf(address(bullStrategy)));
+    }
+
+    function _calcWethToWithdraw(uint256 _bullAmount) internal view returns (uint256) {
+        return _bullAmount.wmul(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy))).wdiv(bullStrategy.totalSupply());
```

Comment:
not sure this will always match due to rounding

contract does the share calculation first, which is ```_bullAmount.wdiv(bullStrategy.totalSupply())```, where this does the multiplication first